scripts/upgrade/validators/verifier-validators-upgrade.ts

```diff
@@ -0,0 +1,149 @@
+import { DeployHelper } from "../../../helpers/DeployHelper";
+import hre, { ethers } from "hardhat";
+import {
+  getConfig,
+  removeLocalhostNetworkIgnitionFiles,
+  verifyContract,
+} from "../../../helpers/helperUtils";
+import { contractsInfo, VALIDATOR_TYPES } from "../../../helpers/constants";
+import fs from "fs";
+import path from "path";
+
+// In real upgrade, you should use THE NAME and THE ADDRESS
+// of your custom verifier contract
+const verifierContractName = "<verifier contract name>";
+const verifierContractAddress = "<verifier contract address>";
+
+const removePreviousIgnitionFiles = false;
+const impersonate = false;
+
+const config = getConfig();
+
+const chainId = hre.network.config.chainId;
+const network = hre.network.name;
+
+async function getSigners(useImpersonation: boolean): Promise<any> {
+  if (useImpersonation) {
+    const proxyAdminOwnerSigner = await ethers.getImpersonatedSigner(config.ledgerAccount);
+    return { proxyAdminOwnerSigner };
+  } else {
+    const [signer] = await ethers.getSigners();
+    const proxyAdminOwnerSigner = signer;
+    return { proxyAdminOwnerSigner };
+  }
+}
+
+async function main() {
+  const verifierContract = await ethers.getContractAt(
+    verifierContractName,
+    verifierContractAddress,
+  );
+
+  if (!ethers.isAddress(config.ledgerAccount)) {
+    throw new Error("LEDGER_ACCOUNT is not set");
+  }
+
+  const { proxyAdminOwnerSigner } = await getSigners(impersonate);
+  console.log("Proxy Admin Owner Address: ", await proxyAdminOwnerSigner.getAddress());
+  if (removePreviousIgnitionFiles) {
+    removeLocalhostNetworkIgnitionFiles(network, chainId);
+  }
+
+  const deployHelper = await DeployHelper.initialize([proxyAdminOwnerSigner], true);
+
+  const countRequests = await verifierContract.getZKPRequestsCount();
+  const validators: {
+    circuitId: string;
+    validatorContractName: string;
+    validatorType: any;
+    validatorContractAddress: string;
+    validatorVerification: any;
+  }[] = [];
+
+  if (countRequests > 0) {
+    const requests = await verifierContract.getZKPRequests(0, countRequests);
+    console.log("Requests found: ", requests.length);
+    for (const request of requests) {
+      if (!validators.find((v) => v.validatorContractAddress === request[1])) {
+        let validatorVerification;
+        let validatorContractName;
+        let validatorType;
+        switch (JSON.parse(request[0]).body.scope[0].circuitId) {
```

Comment:
This will not work if the request JSON is missing or, even worse, wrongly set. Consider getting circuit name from `getSupportedCircuitIds()[0]` of `ICircuitValidator `